src/bessel.jl

```diff
@@ -7,37 +7,37 @@ struct AmosException <: Exception
 end
 
 ## Airy functions
-let
-    const ai::Array{Float64,1} = Array{Float64}(2)
-    const ae::Array{Int32,1} = Array{Int32}(2)
-    global _airy, _biry
-    function _airy(z::Complex128, id::Int32, kode::Int32)
-        ccall((:zairy_,openspecfun), Void,
-              (Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32},
-               Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32}),
-              &real(z), &imag(z),
-              &id, &kode,
-              pointer(ai,1), pointer(ai,2),
-              pointer(ae,1), pointer(ae,2))
-        if ae[2] == 0 || ae[2] == 3
-            return complex(ai[1],ai[2])
-        else
-            throw(AmosException(ae[2]))
-        end
+function _airy(z::Complex128, id::Int32, kode::Int32)
+    ai1, ai2 = Ref{Float64}(), Ref{Float64}()
+    ae1, ae2 = Ref{Int32}(), Ref{Int32}()
+
+    ccall((:zairy_,openspecfun), Void,
+          (Ref{Float64}, Ref{Float64}, Ref{Int32}, Ref{Int32},
+           Ptr{Float64}, Ptr{Float64}, Ptr{Int32}, Ptr{Int32}),
```

Comment:
There doesn't seem to be any difference in doing so, but will change since it seems more consistent 

```julia
julia> f(x::Float64) = Base.unsafe_convert(Ref{Float64}, Base.cconvert(Ref{Float64},Ref{Float64}(x))) f (generic function with 1 method)

julia> g(x::Float64) = Base.unsafe_convert(Ptr{Float64}, Base.cconvert(Ptr{Float64},Ref{Float64}(x))) g (generic function with 1 method)

julia> @code_llvm f(2.)

; Function f
; Location: REPL[18]
; Function Attrs: uwtable
define double* @julia_f_63230(double) #0 {
top:
  %1 = alloca i64, align 8
  %2 = bitcast i64* %1 to i8*
  %3 = bitcast i64* %1 to %jl_value_t*
  call void @llvm.lifetime.start(i64 8, i8* %2)
; Location: REPL[18]:1
  %4 = bitcast %jl_value_t* %3 to double*
  ret double* %4
}

julia> @code_llvm g(2.)

; Function g
; Location: REPL[19]
; Function Attrs: uwtable
define double* @julia_g_63231(double) #0 {
top:
  %1 = alloca i64, align 8
  %2 = bitcast i64* %1 to i8*
  %3 = bitcast i64* %1 to %jl_value_t*
  call void @llvm.lifetime.start(i64 8, i8* %2)
; Location: REPL[19]:1
  %4 = bitcast %jl_value_t* %3 to double*
  ret double* %4
}